lib/core/src/Cardano/Wallet/CoinSelection/Internal/Balance.hs

```diff
@@ -816,32 +814,38 @@ performSelection = performSelectionEmpty performSelectionNonEmpty
 --          selectionHasValidSurplus constraints (transformResult result)
 --
 performSelectionEmpty
-    :: Functor m => PerformSelection m NonEmpty u -> PerformSelection m [] u
+    :: forall m ctx. (Functor m, SelectionContext ctx)
+    => PerformSelection m NonEmpty ctx
+    -> PerformSelection m []       ctx
 performSelectionEmpty performSelectionFn constraints params =
     fmap transformResult <$>
     performSelectionFn constraints (transformParams params)
   where
     transformParams
-        :: SelectionParamsOf []       u
-        -> SelectionParamsOf NonEmpty u
-    transformParams
-        = over #extraCoinSource
-            (transform (`Coin.add` minCoin) (const id))
-        . over #outputsToCover
-            (transform (const (dummyOutput :| [])) (const . id))
+        :: SelectionParamsOf []       ctx
+        -> SelectionParamsOf NonEmpty ctx
+    transformParams p@SelectionParams {..} = p
+        { extraCoinSource =
+            transform (`Coin.add` minCoin) (const id) extraCoinSource
+        , outputsToCover =
+            transform (const (dummyOutput :| [])) (const . id) outputsToCover
+        }
```

Comment:
I changed this section of code to use ordinary record update syntax rather than using `over` (from `generic-lens`).

For some reason that I'm currently unable to ascertain, `over` doesn't seem to play well with type families. I'll investigate changing this back in a future PR.